staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -95,6 +95,26 @@ type Reflector struct {
 	// etcd, which is significantly less efficient and may lead to serious performance and
 	// scalability problems.
 	WatchListPageSize int64
+	// Called whenever the ListAndWatch drops the connection with an error.
+	watchErrorHandler WatchErrorHandler
+}
+
+type WatchErrorHandler func(r *Reflector, err error)
```

Comment:
I'd repeat your disclaimer (about not doing work that will take time) in a comment here, as well.

Reply:
done

Reply:
Should the caller put a cap on the amount of time the handler executes ?
Just in case some implementation doesn't follow the guide above.

Reply:
@tedyu can you point to an established pattern for doing that safely? were you thinking a done channel to the handler, or something else?

Reply:
Yeah, similar to done channel.
You can create goroutine for running the handler.
In the caller, use a timer to gauge the amount of time handler runs.

Reply:
I think that is too much complexity for this. If people abuse this, we can fix it. Until then, just documentation, please.